js/layout-javascript/news-feed-code.js

```diff
@@ -1980,9 +1980,12 @@ DynamicList.prototype.getEntryComments = function(options) {
           }
         });
       })
-      .then(function(entries){
-        record.comments = entries;
-        record.commentCount = entries.length;
+      .then(function(entries) {
+        var filteredEntries = _.filter(entries, function(entry) {
+          return entry.data.settings.text.trim();
```

Comment:
We always like to be more careful with objects in case content is missing. The following allows any missing properties to return an empty string as a result, preventing errors.

```suggestion
          return _.get(entry, 'data.settings.text', '').trim();
```

Though, I think we can use queries to offload this to the backend. See my other comment.

Reply:
I couldn't comment on unmodified code. See a few lines up where a query object is provided. I'd change:

```js
instance.query({
  allowGrouping: true,
  where: {
    content: content
  }
})
```

...to...

```js
instance.query({
  allowGrouping: true,
  where: {
    content: content,
    settings: {
      text: { $regex: '[^\s]*' }
    }
  }
})
```

This is untested, but I imagine this will update the query so that it excludes empty text values right away. Use the developer network tool to confirm.

Reply:
It works